src/Core/Framework/DataAbstractionLayer/Event/BeforeVersionMergeEvent.php

```diff
@@ -0,0 +1,39 @@
+<?php declare(strict_types=1);
+
+namespace Shopware\Core\Framework\DataAbstractionLayer\Event;
+
+use Shopware\Core\Framework\Log\Package;
+use Symfony\Contracts\EventDispatcher\Event;
+
+/**
+ * @phpstan-type WriteOperation array<string, array<int, mixed>>
+ * @phpstan-type Writes array{
+ *     insert: WriteOperation,
+ *     update: WriteOperation,
+ *     delete: WriteOperation
+ * }
+ */
+#[Package('core')]
+class BeforeVersionMergeEvent extends Event
```

Comment:
Please don't use references here. Just expose a public property and use the result in the version manager. If you want to be fancy you could expose a filter method something like:


```php
public function filterWrites(callable $callback): array
{
    return array_filter($this->writes, $callback);
}
```